frontend/src/custom-hooks/useAuthzStakingExecHelper.tsx

```diff
@@ -166,18 +209,32 @@ const useAuthzStakingExecHelper = () => {
     const address = convertAddress(data.chainID, data.granter);
     const grants: Authorization[] =
       authzChains?.[data.chainID]?.GrantsToMeAddressMapping?.[address] || [];
+    let isExpired = false;
     const haveGrant = grants.some((grant) => {
-      return (
-        // todo: stake Authorization
+      if (
         (grant.authorization['@type'] ===
```

Comment:
I see most of the places this logic being used. It would be better to write write utility to function to rewrite it.
```suggestion
        const utilityFun = (grant, msg1, msg2, ....., dispatch) {
           if (grant.authorization['@type'] === msg1 &&  grant.authorization.msg === msg2 ......) {
               if (isTimeExpired(grant.expiration)){
                  dispatch("permission denied")
               }
           }
        }
```